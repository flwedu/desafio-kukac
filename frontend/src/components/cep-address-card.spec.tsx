import { render } from "@testing-library/react";
import CepAddressCard from "./cep-address-card";
describe("Cep Address Card components tests", () => {
  test("Should not render the component if address.cep is invalid", () => {
    const data = {};
    const { container } = render(<CepAddressCard data={data} />);

    expect.assertions(2);
    expect(container.querySelector("#cep-card")).toBeFalsy();
    expect(container).toMatchSnapshot();
  });

  test("Should render the component properly", () => {
    const data = {
      cep: "44900-000",
      localidade: "IrecÃª",
      uf: "BA",
    };
    const { container } = render(<CepAddressCard data={data} />);

    expect.assertions(2);
    expect(container.querySelector("#cep-card")).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});

export {};
