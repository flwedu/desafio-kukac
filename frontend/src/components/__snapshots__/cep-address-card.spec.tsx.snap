// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cep Address Card components tests Should not render the component if address.cep is invalid 1`] = `<div />`;

exports[`Cep Address Card components tests Should render the component properly 1`] = `
<div>
  <div
    class="cep-card"
    id="cep-card"
  >
    <div
      class="header"
    >
      <h1>
        CEP: 
        44900-000
      </h1>
    </div>
    <div
      class="body"
    >
      <div
        class="field"
      >
        Cidade
        : 
        IrecÃª
      </div>
      <div
        class="field"
      >
        UF
        : 
        BA
      </div>
    </div>
  </div>
</div>
`;
